test/widgets/chat/drive_chat_page_test.dart

```diff
@@ -0,0 +1,333 @@
+import 'package:motis_mitfahr_app/account/models/profile.dart';
+import 'package:motis_mitfahr_app/account/widgets/avatar.dart';
+import 'package:motis_mitfahr_app/drives/models/drive.dart';
+import 'package:motis_mitfahr_app/drives/pages/drive_chat_page.dart';
+import 'package:motis_mitfahr_app/rides/models/ride.dart';
+import 'package:motis_mitfahr_app/util/chat/models/message.dart';
+import 'package:motis_mitfahr_app/util/chat/pages/chat_page.dart';
+import 'package:motis_mitfahr_app/util/supabase.dart';
+import 'package:supabase_flutter/supabase_flutter.dart';
+
+import '../../util/factories/chat_factory.dart';
+import '../../util/factories/drive_factory.dart';
+import '../../util/factories/message_factory.dart';
+import '../../util/factories/model_factory.dart';
+import '../../util/factories/profile_factory.dart';
+import '../../util/factories/ride_factory.dart';
+import '../../util/mock_server.dart';
+import '../../util/pump_material.dart';
+import '../../util/request_processor.dart';
+import '../../util/request_processor.mocks.dart';
+
+void main() {
+  late Drive drive;
+  final MockRequestProcessor processor = MockRequestProcessor();
+  final Profile profile = ProfileFactory().generateFake(id: 1);
+
+  setUpAll(() {
+    MockServer.setProcessor(processor);
+  });
+
+  group('Has Stream Subscription', () {
+    setUp(() async {
+      whenRequest(processor).thenReturnJson([]);
+    });
+    testWidgets('when Drive has rides with active chat', (WidgetTester tester) async {
+      drive = DriveFactory().generateFake(
+        driverId: profile.id,
+        rides: [RideFactory().generateFake(chatId: 1, status: RideStatus.approved)],
+      );
+      await pumpMaterial(tester, DriveChatPage(drive: drive));
+      await tester.pump();
+      verifyRequest(
+        processor,
+        urlMatcher: equals('/rest/v1/messages?select=%2A&order=created_at.desc.nullslast'),
+      ).called(1);
+      final List<RealtimeChannel> subscription = SupabaseManager.supabaseClient.getChannels();
+      expect(subscription.length, 1);
+      expect(subscription[0].topic, 'realtime:public:messages:1');
+    });
+
+    testWidgets('not when Drive has no rides', (WidgetTester tester) async {
+      drive = DriveFactory().generateFake(
+        driverId: profile.id,
+        rides: [],
+      );
+      await pumpMaterial(tester, DriveChatPage(drive: drive));
+      await tester.pump();
+      verifyRequestNever(
+        processor,
+        urlMatcher: equals('/rest/v1/messages?select=%2A&order=created_at.desc.nullslast'),
+      );
+      final List<RealtimeChannel> subscription = SupabaseManager.supabaseClient.getChannels();
+      expect(subscription.length, 0);
+    });
+
+    testWidgets('not when Drive has rides without active chat', (WidgetTester tester) async {
+      drive = DriveFactory().generateFake(
+        driverId: profile.id,
+        rides: [RideFactory().generateFake(status: RideStatus.pending)],
```

Comment:
maybe also add a ride with another status to this array, just to make the point more clear.

Reply:
yes i can do that i thought that the point is clear enough because the activeChat method is there and tested